x-pack/lib/monitoring/inputs/metrics/stats_event_factory.rb

```diff
@@ -124,8 +125,13 @@ def format_queue_stats(agent, stats)
     end
 
     def fetch_node_stats(agent, stats)
+      if @webserver_enabled
+        http_addr = stats.get_shallow(:http_address).value
+      else
+        http_addr = "#{LogStash::WebServer::DEFAULT_HOST}:#{LogStash::WebServer::DEFAULT_PORTS.first}"
```

Comment:
I believe we can set this to null when the webserver is disabled, to avoid recording misleading information.
A null should be well accepted in elasticsearch, so I don't foresee major issues from this change, other than some tests failing

Reply:
I configured locally Logstash to send data to Elasticsearch and when we don't put the `http_addr` in the document we send to the monitoring ES, in Kibana it doens't show anything about the monitored LS. I started ES cleaning the `.monitoring-logstash-7` indexes before the test

Reply:
In this case we should make the logstash monitoring plugin in Kibana work well with documents that don't require the http address to be an actual value

Reply:
I'll look into

Reply:
I've retried, if we don't send `http_address`  then no data is shown *but* if we send it null, then in Kibana we can see the Logstash monitoring dashboards, so I go with null